### PR TITLE
Upgrade Rails from 6 to 7

### PR DESCRIPTION
**Pull request by Gargron**
[Instructions](https://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-6-1-to-rails-7-0)

Some gems require extra upgrade work:

- [ ] nsa: (No upgrade available)
- [ ] devise-two-factor: [Instructions](https://github.com/tinfoil/devise-two-factor/blob/main/UPGRADING.md#upgrading-from-4x-to-5x)

___

Fixes MAS-101

### CODE REVIEW

**Comment by nschonni** [comment]
I think there is at least on JavaScript dep that needs to bump with it that I saw in https://github.com/mastodon/mastodon/pull/23966

File: config/initializers/new_framework_defaults_7_0.rb
Comment (mjankowski) [inline config/initializers/new_framework_defaults_7_0.rb:1]: Unless we actually need any of these overrides, you can delete this file.

File: config/initializers/new_framework_defaults_7_0.rb [hunk 1]
```diff
@@ -0,0 +1,135 @@
+# Be sure to restart your server when you modify this file.
+#
+# This file eases your Rails 7.0 framework defaults upgrade.
+#
+# Uncomment each configuration one by one to switch to the new default.
+# Once your application is ready to run with all new defaults, you can remove
+# this file and set the `config.load_defaults` to `7.0`.
```
Comment (mjankowski) [inline config/initializers/new_framework_defaults_7_0.rb:7]: The exception to my prior comment being to do this first in application.rb!

File: Gemfile [hunk 2]
```diff
@@ -61,7 +61,7 @@ gem 'kaminari', '~> 1.2'
 gem 'link_header', '~> 0.0'
 gem 'mime-types', '~> 3.4.1', require: 'mime/types/columnar'
 gem 'nokogiri', '~> 1.14'
-gem 'nsa', '~> 0.2'
+# gem 'nsa', '~> 0.2'
```
Comment (mjankowski) [inline Gemfile:64]: This gem looks not maintained ... the last changes were from you almost 3 years ago working on rails 6 support.

Should we do that again? Are there alternate gems or other logging approaches which could fill the same need?

**Comment by Gargron** [comment]
Closing this in favour of work in #24241

**Pull request closed without merging**